t/10-artifact-resolve.t

```diff
@@ -15,29 +16,51 @@ my $ar = Net::SAML2::Protocol::ArtifactResolve->new(
 
 isa_ok($ar, 'Net::SAML2::Protocol::ArtifactResolve');
 
-my $override
-    = Sub::Override->override(
-    'Net::SAML2::Protocol::ArtifactResolve::issue_instant' =>
-        sub { return 'myissueinstant' });
+my $override = Sub::Override->override(
+    'Net::SAML2::Protocol::ArtifactResolve::issue_instant' => sub {
+        return 'myissueinstant';
+    }
+);
 
 $override->override(
-    'Net::SAML2::Protocol::ArtifactResolve::id' => sub { return 'myid' });
+    'Net::SAML2::Protocol::ArtifactResolve::id' => sub { return 'myid' }
+);
 
 my $xml = $ar->as_xml;
 
 my $xp = get_xpath(
     $xml,
-    samlp => 'urn:oasis:names:tc:SAML:2.0:protocol',
-    saml  => 'urn:oasis:names:tc:SAML:2.0:assertion',
+    samlp => URN_PROTOCOL,
+    saml  => URN_ASSERTION,
 );
 
 test_xml_attribute_ok($xp, '/samlp:ArtifactResolve/@ID', 'myid');
 test_xml_attribute_ok($xp, '/samlp:ArtifactResolve/@IssueInstant',
     'myissueinstant');
 
 test_xml_value_ok($xp, '/samlp:ArtifactResolve/samlp:Artifact',
-    'some-artifact',);
+    'some-artifact');
 
 test_xml_attribute_ok($xp, '/samlp:ArtifactResolve/@ProviderName',
-    'Net-SAML2 Test',);
+    'Net-SAML2 Test');
+
+{
+    my $ar = Net::SAML2::Protocol::ArtifactResolve->new(
+        issuer      => 'http://some/sp',
+        destination => 'http://some/idp',
+        artifact    => 'some-artifact',
+    );
+
+    my $xml = $ar->as_xml;
+    my $xp = get_xpath(
+        $xml,
+        samlp => URN_PROTOCOL,
+        saml  => URN_ASSERTION,
+    );
+
+    my $nodes = $xp->findnodes('/samlp:ArtifactResolve/@ProviderName');
+    is($nodes->size, 0, "We don't have a provider name");
```

Comment:
reviewing on my phone.   Do we have a test for a provider if it is provided?

Reply:
Yes we have, the tests start with provider 'Net-SAML2 Test' and on lines 41-42 we test for the attribute being 'Net-SAML2 Test'.